cdap-app-templates/cdap-etl/hydrator-test/src/main/java/io/cdap/cdap/etl/mock/spark/streaming/MockStreamingEventSource.java

```diff
@@ -0,0 +1,213 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.apache.spark.streaming.api.java.JavaDStream;
+
+/**
+ * Mock streaming source that uses a DStream that implements {@link StreamingEventHandler}
+ */
+@Plugin(type = StreamingSource.PLUGIN_TYPE)
+@Name("MockStreamingEventSource")
+public class MockStreamingEventSource extends StreamingSource<StructuredRecord>
+    implements StreamingStateHandler {
+
+  public static final PluginClass PLUGIN_CLASS = getPluginClass();
+  private static final Gson GSON = new Gson();
+  private static final Type STRING_LIST_TYPE = new TypeToken<List<String>>() {
+  }.getType();
+  public static final String MOCK_STREAMING_EVENT_SOURCE_COMPLETED =
+      "MockStreamingEventSource-completed";
+  public static final String MOCK_STREAMING_EVENT_SOURCE_STARTED =
+      "MockStreamingEventSource-started";
+  public static final String MOCK_STREAMING_EVENT_SOURCE_RETRIED =
+      "MockStreamingEventSource-retried";
+
+  private static StreamingEventHandlerTracker tracker = new StreamingEventHandlerTracker();
+
+  private final Conf conf;
+
+  public MockStreamingEventSource(Conf conf) {
+    this.conf = conf;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    try {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(Schema.parseJson(conf.schema));
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Could not parse schema " + conf.schema);
+    }
+  }
+
+  @Override
+  public void prepareRun(StreamingSourceContext context) throws Exception {
+    Schema schema = Schema.parseJson(conf.schema);
+
+    if (conf.referenceName != null && conf.schema != null) {
+      context.registerLineage(conf.referenceName, schema);
+      Schema outputSchema = Schema.parseJson(conf.schema);
+      if (outputSchema.getFields() != null) {
+        outputSchema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList())
+            .forEach(field -> context.record(Collections.singletonList(
+                new FieldReadOperation("Read " + field, "Read from mock source",
+                    EndPoint.of(context.getNamespace(), conf.referenceName), field))));
+      }
+    }
+  }
+
+  @Override
+  public JavaDStream<StructuredRecord> getStream(StreamingContext context) throws Exception {
+    Schema schema = Schema.parseJson(conf.schema);
+    List<String> recordsAsStrings = new Gson().fromJson(conf.records, STRING_LIST_TYPE);
+    return new JavaDStream<>(
+        new MockStreamingEventHandlerDStream(context.getSparkStreamingContext().ssc(),
+            recordsAsStrings, schema, tracker),
+        scala.reflect.ClassTag$.MODULE$.apply(String.class));
+  }
+
+  /**
+   * Config for mock source.
+   */
+  private static class Conf extends PluginConfig {
+
+    private String schema;
+    private String records;
+    @Nullable
+    private Long intervalMillis;
```

Comment:
this looks unused, unless I'm missing something?

Reply:
Removed all unused config properties.